### PR TITLE
Hierarchy endpoint

### PR DESCRIPTION
**Pull request by ibilal**
hierarchy endpoint with tests validating GET was successful, and renders hierarchy view successfully with appropriate elements.

### CODE REVIEW

**Comment by RickyV33** [comment]
Note: needs to be merged in before #108

File: test/routes/hierarchy.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          done();
+        });
+        it('should render view with title and unordered list', function (done) {
+          // Render the view using ejs
+          let path = join(__dirname, '../../views/hierarchy.ejs');
+          let data = {title: 'Select a Root Item (Optional) ',
+            itemHierarchy: './mockHierarchy.json'};
+          let renderedView = ejs.compile(read(path, 'utf8'), {filename: path})(data);
+          expect(res.text).to.equal(renderedView);
+          expect(res.text).contains('Select a Root Item (Optional) ');
```
Comment (HansenML) [inline test/routes/hierarchy.js]: If res.text is the same as the rendered view, I don't think you need to make the checks in "contains" on lines 44 and 45. If there is any difference between the rendered view and the res.text it should come up in that very first expect.

File: views/hierarchy.ejs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    <link rel='stylesheet' href='/stylesheets/style.css'/>
+  </head>
+  <body>
+    <h1><%= title %></h1>
+    <h3>Please select a root item from below, and click 'Render Graph': </h3>
+    <ul id="itemHierarchy">
+      <%if(itemHierarchy.length > 0) {%>
+        <%itemHierarchy.forEach(function (item) {%>
+          <li><%= item.name %></li>
+        <%});%>
```
Comment (HansenML) [inline views/hierarchy.ejs]: You have extra <%> containers here, they ignore line breaks. So you only need them to start and end an entire block of JS, not a single line. So the end of 11 and beginning of 12 can be removed and so can the end of 14 and beginning of 15. Not that it changes anything just cleans it up a bit. If you don't have any other big issues to fix, I'd ignore this one.

File: package.json [hunk 3]
```diff
@@ -8,8 +8,9 @@
     "test": "semistandard | snazzy && mocha --check-leaks --recursive",
     "cover": "istanbul cover _mocha -- --recursive",
     "coveralls": "istanbul cover _mocha --report lcovonly -- --recursive && cat ./coverage/lcov.info | coveralls",
-    "build": "npm run build:index",
+    "build": "npm run build:index & build:hierarchy",
```
Comment (kathtran) [inline package.json]: `"build": "npm run build:index & npm run build:hierarchy",`

File: test/routes/hierarchy.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          children: [{name: 'Nested child of root 1', children: []}]}]},
+      {name: 'Root item 2',
+        children: [{name: 'Root 2 Child 1', children: []}, {name: 'Root 2 Child 2',
+          children: [{name: 'Nested child of root 2', children: []}]}]}]}
+];
+
+describe('hierarchy', function () {
+  describe('GET /hierarchy', function () {
+    it('Should render item hierarchy view according to item hierarchy test case', function (done) {
+      hierarchyTestCases.forEach(function (item) {
```
Comment (chances) [inline test/routes/hierarchy.js]: Should each test case be it's own `it`, like the auth tests?

**Pull request merged**